internal/schemautil/custom_diff.go

```diff
@@ -116,6 +116,16 @@ func CustomizeDiffCheckPlanAndStaticIpsCannotBeModifiedTogether(_ context.Contex
 	return nil
 }
 
+// CustomizeDiffCheckStaticIpsExistence prevents users from modifying `aiven_static_ip` resources in
+// the same step with the resources that uses them due to Terraform limitation
+// See: https://terraform.io/plugin/sdkv2/resources/customizing-differences
+func CustomizeDiffCheckStaticIpsExistence(_ context.Context, d *schema.ResourceDiff, _ interface{}) error {
+	if _, exists := d.GetOkExists("static_ips"); d.Id() != "" && !exists {
```

Comment:
Also we can replace `d.Id() != ""` to `!d.IsNewResource`, perhaps it is more verbose in the IF statement.